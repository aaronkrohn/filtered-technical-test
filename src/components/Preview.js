import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import AssetList from './AssetList';

const Wrapper = styled.div`
  flex: 0 0 67%;
  max-width: 67%;
  min-height: 0.063em;
  padding: 0 1em;
`;

const Header = styled.h1`
  font-size: 1.5em;
  margin: 0 0 0.5em 0;
`;

const Preview = ({ assetList }) => (
  <Wrapper>
    <Header>Learning Assets Preview</Header>
    <AssetList assetList={assetList} />
  </Wrapper>
);

Preview.propTypes = {
  assetList: PropTypes.arrayOf(PropTypes.object)
};

export default Preview;
