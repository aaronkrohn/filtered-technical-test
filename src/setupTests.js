import React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducers from './reducers/index';

export const renderWithRedux = (
  ui,
  { initialState, store = createStore(reducers, initialState) } = {}
) => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store
});
