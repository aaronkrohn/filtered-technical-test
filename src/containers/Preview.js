import { connect } from 'react-redux';
import Preview from '../components/Preview';

const mapStateToProps = ({ previewReducer }) => ({
  assetList: previewReducer.assetList
});

export default connect(mapStateToProps)(Preview);
