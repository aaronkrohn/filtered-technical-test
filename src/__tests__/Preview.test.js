import React from 'react';
import { render } from 'react-testing-library';
import { renderWithRedux } from '../setupTests';
import Preview from '../components/Preview';
import ConnectedPreview from '../containers/Preview';

const mockAsset = {
  id: 99,
  name: 'Mock Name',
  type: 'Mock Type',
  duration: '0.11',
  durationType: 'min'
};

test('renders', () => {
  const { container } = render(<Preview />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders with Redux', () => {
  const { queryByText } = renderWithRedux(<ConnectedPreview />);
  expect(queryByText('10 Principles of Change Management')).toBeNull();
});

test('renders when as asset has been moved from the sidebar', () => {
  const { container } = renderWithRedux(<ConnectedPreview />, {
    initialState: { previewReducer: { assetList: [mockAsset] } }
  });
  expect(container.querySelector('h2')).toHaveTextContent('Mock Name');
});
