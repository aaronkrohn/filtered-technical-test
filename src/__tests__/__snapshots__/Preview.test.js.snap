// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  class="sc-EHOje lpawV"
>
  <h1
    class="sc-bZQynM gIPWju"
  >
    Learning Assets Preview
  </h1>
  <ul
    class="sc-ifAKCX hcmkp"
  />
</div>
`;
